src/main/java/org/jabref/logic/util/io/FileUtil.java

```diff
@@ -174,19 +171,19 @@ public static Optional<String> getUniquePathFragment(List<String> paths, Path co
      * @return the minimal unique path substring for each file path
      */
     public static List<String> uniquePathSubstrings(List<String> paths) {
-        List<Stack<String>> stackList = new ArrayList<>(paths.size());
+        List<Deque<String>> stackList = new ArrayList<>(paths.size());
```

Comment:
Stack iteration returns in a wrong different order than deque that's why the test fails.

Reply:
Might have the same effect in the other places

Reply:
> Stack iteration returns in a wrong different order than deque that's why the test fails.

According to JDK not:

![image](https://github.com/JabRef/jabref/assets/1366654/4c516b7a-f3e3-4a90-95f4-1edae3e3c87b)

I did not use the dequeue methods, but sticked to the stack ones:

![image](https://github.com/JabRef/jabref/assets/1366654/2e2346c8-e1b4-40ba-83fd-12f86d61763b)

![image](https://github.com/JabRef/jabref/assets/1366654/02f19a3d-fbe4-4e12-ba61-33146a1a4769)

This is so strange, I will debug now.

Reply:
JavaDoc: https://docs.oracle.com/javase/8/docs/api/java/util/Deque.html#push-E-

Reply:
see https://stackoverflow.com/a/54536611/3450689
https://stackoverflow.com/a/74487335/3450689